#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material2 "slippy"
      coulombFriction [
        0
      ]
    }
    ContactProperties {
      material2 "block"
      coulombFriction [
        0.01
      ]
    }
  ]
}
Viewpoint {
  orientation -0.8538394353935587 0.450497304081276 0.26078803190006855 0.8639224422957857
  position 0.21397286505872543 0.44495436205179173 0.4215340164468149
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
Floor {
  size 2.4 2.4
  appearance Asphalt {
  }
}
Robot {
  children [
    Compass {
    }
    GPS {
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.025 0.035 0.06
      }
      device [
        RotationalMotor {
          name "pincer_motor_1"
        }
      ]
      endPoint Solid {
        translation 0.025 0.035 0.09
        children [
          Group {
            children [
              Shape {
                appearance PBRAppearance {
                }
                geometry Box {
                  size 0.003 0.05 0.06
                }
              }
            ]
          }
        ]
        name "solid(4)"
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.115 0
      }
      device [
        RotationalMotor {
          name "turret_motor"
        }
      ]
      endPoint Solid {
        translation 0 0.115 0
        children [
          Group {
            children [
              DistanceSensor {
                translation 0 -0.09 0.125
                rotation 0 1 0 -1.5708
                name "distance_sensor_front"
                lookupTable [
                  0 0 0
                  0.8 1000 0
                ]
              }
              Transform {
                translation 0 -0.05 0.12
                children [
                  Shape {
                    appearance PBRAppearance {
                    }
                    geometry Box {
                      size 0.01 0.1 0.01
                    }
                  }
                ]
              }
              Shape {
                appearance PBRAppearance {
                }
                geometry Box {
                  size 0.01 0.01 0.25
                }
              }
            ]
          }
        ]
        name "solid(3)"
      }
    }
    Solid {
      translation 0 0.01 -0.12
      children [
        DEF caster_wheel Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.5
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "solid(2)"
      contactMaterial "slippy"
      boundingObject USE caster_wheel
      physics Physics {
      }
    }
    LightSensor {
      rotation 0 1 0 -1.57
      lookupTable [
        0 0 0
        4 4 0
      ]
    }
    DEF body Transform {
      translation 0 0.06 -0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.15 0.1 0.2
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.08 0.05 0
      }
      device [
        RotationalMotor {
          name "drive_motor_2"
        }
      ]
      endPoint Solid {
        translation -0.0799999992169501 0.050000255479939366 -1.165132989193784e-05
        rotation 0.0008216087360406806 -0.0008216119450159905 -0.9999993249562206 4.712384632093233
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.05
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.08 0.05 0
      }
      device [
        RotationalMotor {
          name "drive_motor_1"
        }
      ]
      endPoint Solid {
        translation 0.08000000078305933 0.05000025763117322 -1.1651286625896926e-05
        rotation 0.0008180946623314349 -0.0008180978583993661 -0.9999993307182847 4.712384637855301
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.05
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "test_controller"
}
target {
  translation -0.46 0.04 0.75
  name "target(3)"
}
target {
  translation -0.64 0.05 -0.42
  name "target(2)"
}
target {
  translation 0.45 0.05 0.38
  name "target(1)"
}
target {
  translation 0.65 0.04 0
}
DEF Floor_colour Solid {
  translation 0 0.0001 0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(6)"
}
DEF Floor_colour Solid {
  translation 0 0.0001 -0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(5)"
}
DEF WALL_1 Solid {
  translation 1.2 0.05 0
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0.333333 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 2.38
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -1.2 0.05 0
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
TexturedBackgroundLight {
  texture "empty_office"
}
